src/components/micro-frame-sse/README.md

```diff
@@ -0,0 +1,254 @@
+<h1 align="center">
+  <!-- Logo -->
+  <br/>
+  micro-frame-sse & micro-frame-slot
+	<br/>
+</h1>
+
+<p align="center">
+  SSE stream support for micro-frame
+</p>
+
+# Installation
+
+```console
+npm install @micro-frame/marko
+```
+
+# How it works
+
+Similar to `<micro-frame>`, the package exposes `<micro-frame-sse>` and `<micro-frame-slot>` tags in order to initiate SSE stream and placement of streaming content.
+
+It works the same way as [micro-frame](../../../README.md) when used in browser vs in node server.
+
+## Why
+
+In addition to the benefits from [micro-frame](../../../README.md), `<micro-frame-sse>` allows short living server-sent event to steam content into designated slots within single http request. In another word, `micro-frame-sse` enables multiplexing of `micro-frame`.
+
+Considering SSE is only one kind of streaming formats, `micro-frame` could potentially support all kinds of streaming formats like `json-stream` with a proper `parser`.
+
+![stream explained](stream_explained.jpg)
+
+# Example
+
+`micro-frame-sse` tag is required to put into earlier stage of page rendering (mainly at beginning of the page), who will initiate the SSE request as soon as page starts. `micro-frame-slot` is where content being injected and can be placed at any desired location.
+
+```marko
+<micro-frame-sse
+  src="sse-source"
+  name="unique_name"
+  read(ev) {
+    // assuming data is in JSON format
+    const data = JSON.parse(ev.data);
+    const slotId = data.slotId, html = data.content;
+    return [slotId, html];
+  }
+>
+  <@catch|err|>
+      Uh-oh! ${err.message || err}
+  </@catch>
+</micro-frame-sse>
+
+<micro-frame-slot from="unique_name" slotId="slot_1">
+  <@loading>
+    Loading...
+  </@loading>
+  <@catch|err|>
+      Uh-oh! ${err.message || err}
+  </@catch>
+</micro-frame-slot>
+<div>
+content in between
+</div>
+<ul>
+  <li>
+    <micro-frame-slot from="unique_name" slotId="slot_2" />
+  </li>
+</ul>
+```
+
+# `<micro-frame-sse>` API
+
+## (required) `src`
+
+A path to SSE endpoint.
+
+```marko
+<micro-frame-sse src="sse-source" name="..." read(...) {...} />
+```
+
+## (required) `name`
+
+A unique name for the stream. A page can have multiple streams.
+
+```marko
+<micro-frame-sse src="..." name="stream_name" read(...) {...} />
+```
+
+## (required) `read`
+
+A function to parse the event which returns slotId and streamed content as an array (optionally an ifDone flag). The input is `MessageEvent`, please refer to [MessageEvent](https://developer.mozilla.org/en-US/docs/Web/API/EventSource/message_event#event_properties) for details.
+
+```marko
+<micro-frame-sse src="..." name="..." read(ev) {
+  // logic to fetch slotId and html_content from event
+  return [slotId, html_content];
+
+  // if the ifDone flag is set to true, slot will be closed after reading.
+  // const ifDone = true;
+  // return [slotId, html_content, ifDone];
+} />
+```
+
+## `method`
```

Comment:
I wonder if we should remove `method` and `body` options, and just require using `fetch` if you have those requirements? Since presumably a recommendation would be to avoid anything other than `GET` and to avoid passing `body` content likewise by default.